plugins/org.python.pydev.parser/tests/org/python/pydev/parser/PyParser38Test.java

```diff
@@ -141,4 +141,690 @@ public void testPositionalArgs4() {
         parseLegalDocStr("a = lambda a, b=1, /, c=2, d=3, *, e=4, f=5: None");
     }
 
+    public void testUString() throws Exception {
```

Comment:
The tests don't seem consistent with what was already there... I checked locally and the expected value inside the string shouldn't really include the prefixes.

I'm providing a function below which simplifies the testing a bit (please add the remaining use cases, but note that something as `ur'foo'` returns `foo` as the value, thus, something as `b'foo` should also return just `foo` -- the `Str` ast attributes should be checked for that).

Also, you don't need that many tests (just a few for the corner cases as empty strings/ string with contents) should be ok.

```
    private void checkStr(String s, String expected, boolean binary, boolean fstring, boolean unicode, boolean raw) {
        SequencialASTIteratorVisitor visitor = SequencialASTIteratorVisitor.create(parseLegalDocStr(s));
        List<ASTEntry> asList = visitor.getAsList(Str.class);
        assertEquals(asList.size(), 1);
        ASTEntry entry = asList.iterator().next();
        Str str = (Str) entry.node;
        assertEquals(binary, str.binary);
        assertEquals(fstring, str.fstring);
        assertEquals(unicode, str.unicode);
        assertEquals(raw, str.raw);
        assertEquals(expected, str.s);
    }

    public void testStr() throws Exception {
        checkStr("''", "", false, false, false, false);
        checkStr("'s'", "s", false, false, false, false);

        checkStr("b''", "", true, false, false, false);
        checkStr("b's'", "s", true, false, false, false);

        checkStr("f''", "", false, true, false, false);
        checkStr("f's'", "s", false, true, false, false);
    }
```